dev/publish_bom.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+import yaml
+
+from github import Github
+from github.Gist import Gist
+from github.InputFileContent import InputFileContent
+
+from generate_bom import BomGenerator
+from spinnaker.run import run_quick
+
+SERVICES = 'services'
+VERSION = 'version'
+
+COMPONENTS = [
+  'clouddriver',
+  'deck',
+  'echo',
+  'front50',
+  'gate',
+  'igor',
+  'orca',
+  'rosco',
+  'fiat',
+  'spinnaker-monitoring',
+  'spinnaker'
+]
+
+class BomPublisher(BomGenerator):
+
+  def __init__(self, options):
+    self.__rc_version = options.rc_version
+    self.__bom_dict = {}
+    self.__release_version = options.release_version
+    self.__github_publisher = options.github_publisher
+    self.__changelog_file = options.changelog_file
+    self.__github_token = options.github_token
+    self.__gist_user = options.gist_user
+    self.__github = Github(self.__gist_user, self.__github_token)
+    self.__patch_release = options.patch_release
+    super(BomPublisher, self).__init__(options)
+
+  def __unpack_bom(self):
+    """Load the release candidate BOM into memory.
+    """
+    bom_yaml_string = run_quick('hal versions bom {0} --color false'
+                                .format(self.__rc_version), echo=False).stdout.strip()
+    print bom_yaml_string
+    self.__bom_dict = yaml.load(bom_yaml_string)
+    print self.__bom_dict
+
+  def publish_release_bom(self):
+    """Read, update, and publish a release candidate BOM.
+    """
+    self.__unpack_bom()
+    new_bom_file = '{0}.yml'.format(self.__release_version)
+    self.__bom_dict[VERSION] = self.__release_version
+    self.write_bom_file(new_bom_file, self.__bom_dict)
+    self.publish_bom(new_bom_file)
+    # Re-write the 'latest' Spinnaker version.
+    # TODO(jacobkiefer): Update 'available versions' with Halyard when that feature is ready.
+    self.write_bom_file('latest.yml', self.__bom_dict)
```

Comment:
`latest.yml` is a global alias for the last Spinnaker version that was released by us. It lives in the BOM bucket that Halyard knows about. Since we're publishing the release BOM in this step and making the release official, I think it makes sense to replace the latest version in this step as well.